docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,554 @@
+# Use [OCP templates and] ABPs to install hosted components
+
+## Description
+Currently most of our hosted components have their own repositories where they
+maintain changes and then need to synchronize those changes within their
+respective roles. This can cause cause delays and times where features don't
+match up with what is installed or can be configured.
+
+There is also a concern that copying files from one location to another can
+cause cases where files are out of sync or versions don't align with what is
+installed and what is configured (is what we configured correct for the image
+version we installed).
+
+## Rationale
+As was piloted with the Template Service Broker, a role was created to process
+and configure OCP templates that were synchronized (currently still a manual
+process) from the origin repo where the component resides.
+
+This allowed any changes for the installation to be maintained by the developers
+and keep them from needing to be fluent with Ansible. It also afforded a level
+of abstraction from needing to keep in sync with TSB related changes and then
+translate them to changes within an Ansible role.
+
+However, there is still the gap of ensuring our role configures changes appropriate
+for the version of the component we are installing. Prior to 3.4 with Aggregated
+Logging this was resolved and contained due to using a deployer pod, since everything
+was versioned and contained within an artifact the Integration Services team
+provided. This is something that is addressed again now with the Service Broker,
+APBs are configured in such a way that a containerized install/uninstall is done
```

Comment:
By APBs, I assume you mean, "Ansible Play Books"?

And how does a "containerized install/uninstall ... for the component" solve this problem?

Are we proposing to use templates to create the necessary OCP objects the encode the installation of the component, where the templates are maintained in the origin repository along size the components?

Reply:
APB here refers to [Ansible Playbook Bundle](https://github.com/ansibleplaybookbundle/ansible-playbook-bundle)

Reply:
Adding that url into the references section

Reply:
We are proposing to use a containerized install/uninstall to move all configurations back into their component repositories and also not need to sync these files over. We want to decouple the version of openshift-ansible from the version of the component's configuration files -- this would be all within the APB container so that we can keep the correct component configurations with the proper version of the component. 

Using OCP templates is an independent decision for the teams providing the APBs. It could be useful if the team already provides/develops these templates so they aren't repeating themselves but won't be dictated by this proposal.